system-tests/CUserHeader/whatnumber.h

```diff
@@ -0,0 +1,9 @@
+#if !defined(WHATNUMBER)
```

Comment:
Maybe we should also test the very common `#ifndef THETHING` form. This would require a separate header file.

Reply:
I'm not sure I follow?  I can change the syntax easily enough